File: backend/src/domain/entities/plan.py
import uuid
from datetime import datetime, timezone
from enum import Enum

from pydantic import BaseModel, ConfigDict, Field


class TaskPriority(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class Task(BaseModel):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
    plan_id: uuid.UUID
    title: str = Field(max_length=200)
    description: str = ""
    priority: TaskPriority = TaskPriority.MEDIUM
    estimated_effort: str = Field(default="", max_length=100)
    owner_role: str = Field(default="", max_length=100)
    sort_order: int = 0

    model_config = ConfigDict(from_attributes=True)


class Plan(BaseModel):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
    process_id: uuid.UUID
    summary_md: str = ""
    generated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    tasks: list[Task] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/plan.ts
import { apiRequest } from '../lib/api-client';

export interface Findings {
  process_id: string;
  answers: Record<string, string>;
  free_text: string;
  updated_at: string;
}

export interface UpsertFindingsInput {
  answers: Record<string, string>;
  free_text: string;
}

export interface PlanTask {
  id: string;
  title: string;
  description: string;
  priority: 'low' | 'medium' | 'high';
  estimated_effort: string;
  owner_role: string;
  sort_order: number;
}

export interface Plan {
  process_id: string;
  summary_md: string;
  generated_at: string;
  tasks: PlanTask[];
}

export interface ApiCallOptions {
  token: string | null;
  signal?: AbortSignal;
}

export async function getFindings(
  processId: string,
  { token, signal }: ApiCallOptions,
): Promise<Findings> {
  return apiRequest<Findings>(`/processes/${processId}/findings`, { token, signal });
}

export async function saveFindings(
  processId: string,
  input: UpsertFindingsInput,
  { token, signal }: ApiCallOptions,
): Promise<Findings> {
  return apiRequest<Findings>(`/processes/${processId}/findings`, {
    method: 'PUT',
    body: input,
    token,
    signal,
  });
}

export async function generatePlan(
  processId: string,
  { token, signal }: ApiCallOptions,
): Promise<Plan> {
  return apiRequest<Plan>(`/processes/${processId}/generate-plan`, {
    method: 'POST',
    token,
    signal,
  });
}
